migrate to multiplatform (part 02)

BorzdeG opened the pull request:
migrate to MPP+KotlinDSL (separate #698 - part 01)

!!! **diff after merging #703**

* migrated Kotlin DSL (without Android submodules)
* [FIXED] Licenses not included in pom (#694)
* Added IDEA auto-configuration scripts
* Added [Detekt plugin](https://arturbosch.github.io/detekt/)

arnaudgiuliani commented:
I would like to see other MP proposal about continuing to use Gradle DSL. I'm not 100% sure about this part.

BorzdeG commented:
@arnaudgiuliani 
Added MPP configuration for core modules (everything except Android).
For example, in koin-core, I moved part of the code to the Common block
